articles/howtos/make-template-decorator-ready.md

```diff
@@ -0,0 +1,120 @@
+
+Paste in the namespace and interface name in the `Type FullName` field.
+
+> MyCompany.MyModule.Templates.StartupTemplate.IStartupTemplateContract
+
+Run the Software Factory in Intent Architect and apply the change.
+
+![Apply change](images/make-template-decorator-ready/ApplyExposeDecoratorContractChange.png)
+
+In the `StartupTemlateParial.cs` file, you will notice the following changes being made:
+
+- `IHasDecorators<MyCompany.MyModule.Templates.StartupTemplate.IStartupTemplateContract>` has been added to the class's inheritance list.
+- The definition of that interface has been populated for you:
+
+```csharp
+private ICollection<MyCompany.MyModule.Templates.StartupTemplate.IStartupTemplateContract> _decorators = new List<MyCompany.MyModule.Templates.StartupTemplate.IStartupTemplateContract>();
+
+[IntentManaged(Mode.Fully)]
+public void AddDecorator(MyCompany.MyModule.Templates.StartupTemplate.IStartupTemplateContract decorator)
+{
+    _decorators.Add(decorator);
+}
+
+[IntentManaged(Mode.Fully)]
+public IEnumerable<MyCompany.MyModule.Templates.StartupTemplate.IStartupTemplateContract> GetDecorators()
+{
+    return _decorators;
+}
+```
+
+## Aggregate all the Decorators output
+
+Let's create a method that will aggregate all the Decorator output into a single string in that same class:
+
+```csharp
+private string GetConfigureCode()
+{
+    return GetDecorators()
+                .OrderBy(o => o.Priority)
+                .Aggregate(new StringBuilder(), (sb, dec) => sb.AppendLine(dec.ConfigureCode())).ToString();
+}
+```
+
+## Generate the injected code in Template
+
+Now lets go to the `StartupTemplate.tt` file and make a call to this newly defined method:
+
+```csharp
+// [IntentManaged(Mode.Ignore)] // Uncomment to take over configuring services
+public void Configure(IApplicationBuilder app, IHostingEnvironment env)
+{
+    if (env.IsDevelopment())
+    {
+        app.UseDeveloperExceptionPage();
+    }
+
+    <#= GetConfigureCode() #>
+    app.UseMvc();
+}
+```
+
+This will now allow a Decorator to use the interface being provided to inject new code into this Template.
+
+Recompile this project to ensure the Module is built.
+
+## Re-install that Module in your application
+
+Open up the `Test.App` in Intent Architect, click on the `Modules` on the side panel and then on the `Installed` link at the top of the page.
+
+In the Repository dropdown, select `My Module`.
+
+![Installed Modules](images/make-template-decorator-ready/AppInstalledModules.png)
+
+Locate the `MyCompany.MyModule` and click on the `Reinstall` button located on the right-side panel.
+
+When you click on the Code Generation run, there will be one change that it will attempt to make which is an empty line. This is ok. Just proceed to apply.
+
+This concludes this How-to. If you need to create a Decorator for this Template, please follow the How-to located [here](xref:CreateNewDecorator).
```

Comment:
All my comments further for this article would likely be similar to the previous article I commented on, towards making the article as concise as possible and making the tone quite neutral.

So pending our decision on how we want to handle style and tone in general, I'm not going to comment further on this article for now.